src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1574,6 +1574,7 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
         if (empty($config['transports'])) {
             $container->removeDefinition('messenger.transport.symfony_serializer');
             $container->removeDefinition('messenger.transport.amqp.factory');
+            $container->removeDefinition('messenger.transport.doctrine.factory');
```

Comment:
If the transport lives in the other bundle, then this needs to live in the other bundle too (it will do nothing here, that service doesn't exist in this container builder). But it might not be needed - I would check. But, if messenger is disabled, that service should become an orphan, and because it's private, it'll be removed from the container.

Reply:
Good catch ! In the other bundle the services are loaded only if the messenger component is available.